From 27827456308ca0af5e449930a952b9437b8b690b Mon Sep 17 00:00:00 2001
From: Myles Watson <mylesgw@google.com>
Date: Thu, 3 Dec 2020 18:54:14 -0800
Subject: [PATCH] セキュリティパッチ 2021-03-01 ANDROID-174052148

[porting]

SDP: Only start discovery once

Bug: 174052148
Test: pair with headphones
Tag: #security

CVE-2021-0397

Change-Id: I1d014a7b793bb1b66e26652f6696499ea36a6510
(cherry picked from commit cfa5a74ea90a09e1c7413a25f04332ee2d1e3f21)
Merged-In: I1d014a7b793bb1b66e26652f6696499ea36a6510
(cherry picked from commit f0994f6e4723eddaa617b68139f064d945d9389e)
---
 bta/ag/bta_ag_sdp.cc | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/bta/ag/bta_ag_sdp.cc b/bta/ag/bta_ag_sdp.cc
index ae3c97391..a56f54e08 100644
--- a/bta/ag/bta_ag_sdp.cc
+++ b/bta/ag/bta_ag_sdp.cc
@@ -31,6 +31,7 @@
 #include "bta_api.h"
 #include "bta_sys.h"
 #include "btm_api.h"
+#include "osi/include/log.h"
 #include "osi/include/osi.h"
 #include "sdp_api.h"
 #include "utl.h"
@@ -419,6 +420,12 @@ void bta_ag_do_disc(tBTA_AG_SCB* p_scb, tBTA_SERVICE_MASK service) {
     return;
   }
 
+  if (p_scb->p_disc_db != nullptr) {
+    android_errorWriteLog(0x534e4554, "174052148");
+    APPL_TRACE_ERROR("Discovery already in progress... returning.");
+    return;
+  }
+
   /* allocate buffer for sdp database */
   p_scb->p_disc_db = (tSDP_DISCOVERY_DB*)osi_malloc(BTA_AG_DISC_BUF_SIZE);
   /* set up service discovery database; attr happens to be attr_list len */
-- 
2.30.0.280.ga3ce27912f-goog

